scripts/bdc/xml_generator.py

```diff
@@ -0,0 +1,313 @@
+    study_name.text = metadata_dict.get('Study Name', '')
+    
+    study_name_report = ET.SubElement(config, "StudyNameReportPage")
+    study_name_report.text = metadata_dict.get('Study Name', '')
+    
+    study_types = ET.SubElement(config, "StudyTypes")
+    study_type = ET.SubElement(study_types, "StudyType")
+    study_type.text = "Case Set"
+    
+    description = ET.SubElement(config, "Description")
+    description.text = f"<p>{metadata_dict.get('Description', '')}</p>"
+    
+    program = ET.SubElement(config, "Program")
+    program.text = metadata_dict.get('Program', '')
+    
+    if 'Notes' in metadata_dict and pd.notna(metadata_dict['Notes']):
+        notes = ET.SubElement(config, "Notes")
+        notes.text = metadata_dict['Notes']
+    
+    return root
+
+def create_data_dict_xml(df_group, table_id, study_id, gen3_metadata):
+    if isinstance(gen3_metadata, (dict, pd.Series)):
+        study_version = extract_version_from_accession(gen3_metadata.get('Accession', ''))
+    else:
+        study_version = gen3_metadata
+    
+    root = ET.Element("data_table")
+    root.set("id", table_id)
+    root.set("study_id", f"{study_id}.{study_version}")
+    root.set("participant_set", "1")
+    
+    desc = df_group['columnmeta_var_group_description'].iloc[0] if 'columnmeta_var_group_description' in df_group.columns else ''
+    if pd.notna(desc):
+        description = ET.SubElement(root, "description")
+        description.text = str(desc)
+    
+    for _, row in df_group.iterrows():
+        variable = ET.SubElement(root, "variable")
+        
+        var_id = row.get('varId', '')
+        var_version = 'v1'
+        
+        variable.set("id", f"{var_id}.{var_version}")
+        
+        name = ET.SubElement(variable, "name")
+        name.text = str(row.get('columnmeta_name', ''))
+        
+        desc = ET.SubElement(variable, "description")
+        desc.text = str(row.get('columnmeta_description', ''))
+        
+        type_elem = ET.SubElement(variable, "type")
+        if row.get('is_categorical', False):
+            type_elem.text = "encoded value"
+            if pd.notna(row.get('values', '')):
+                try:
+                    values = eval(str(row['values']))
+                    for value in values:
+                        value_elem = ET.SubElement(variable, "value")
+                        if value == "Male":
+                            value_elem.set("code", "1")
+                        elif value == "Female":
+                            value_elem.set("code", "2")
+                        value_elem.text = str(value)
+                except Exception as e:
+                    logging.warning(f"Error processing values for variable {var_id}: {e}")
+        else:
+            type_elem.text = "string"
+    
+    return root
+
+def save_xml(root, filepath, add_stylesheet=True):
+    xml_str = minidom.parseString(ET.tostring(root)).toprettyxml(indent="    ")
```

Comment:
I'm not sure what the minidom.parseString() is doing here -- shouldn't `.toprettyxml()` already return a string? I think a comment here would be useful.